### PR TITLE
Use minimal content in input data frames

### PR DESCRIPTION
**Pull request by DPeterK**
To speed up execution of the data processing pipeline of the DIMEX R package, we can remove all the columns from the input data frames (population and time use survey data) that aren't directly used by the data processing in the pipeline, and also slim down the columns in the intermediate data frame `activities_complete` that's generated by the data processing pipeline.

### CODE REVIEW

**Review by natalietf — APPROVED** [review]
Looks good to me :) like the removal of all the labels - makes it a lot simpler

**Comment by DPeterK** [comment]
Booo, this change has made the API 3x slower :( Need to dig into this a bit more, I think...

**Comment by DPeterK** [comment]
To explore in more detail why the code has slowed down for cutting down the amount of input data, I've profiled the code with prodigious use of `Sys.time()` calls and `print` statements. The results are as follows:

Step | Full data (s) | Minimised data (s) | Difference (full - minimised) (s)
-----|---------------|--------------------|----------------------------------
Load population data | 9.268 | 9.045 | 0.223 |
Load time use survey data | 8.693 | 8.858 | **+0.165** |
Load PM2.5 data | 2.003 | 1.881 | 0.122 |
Merge PM2.5 data | 3.625 | 3.148 | 0.477 |
Activity sampler run time | 6.329 | 4.706 | 1.623 |
Estimate exposures run time | 29.359 | 148.4 | **+119.041** (5x slower) |
Daily averages run time | 1.187 | 1.066 | 0.121 |
Total execution time | 60.463 | 177.1 | **+116.637** (3x slower) |

Differences **in bold** indicate that running with the minimised data is **slower**.

Note that there are code changes as well between full data and minimised data, to handle the changed sets of column names available in the two datasets.

### Analysis

It's clear that all the slowdown in execution is occurring in the **exposure estimation** section of the pipeline, so why this is occurring is the next thing to explore in more detail. This section alone is 5x slower with minimised data than with full data, leading to a 3x slowdown in overall execution time. Given that, in the API, the data is loaded when the API container is run rather than when the API endpoint is hit, the effective slowdown of the API will be even greater than is suggested by the data here.

**Comment by DPeterK** [comment]
Turns out that the lookups I was doing to get the season label in `calculate_household.R` was responsible for all of the slowdown seen here...

Here's an updated table of run timings:

Step | Full data (s) | Minimised data (s) | Difference (full - minimised) (s)
-----|---------------|--------------------|----------------------------------
Load population data | 9.268 | 8.330 | 0.938 |
Load time use survey data | 8.693 | 8.076 | 0.617 |
Load PM2.5 data | 2.003 | 1.745 | 0.258 |
Merge PM2.5 data | 3.625 | 3.640 | **+0.015** |
Activity sampler run time | 6.329 | 3.999 | 2.330 |
Estimate exposures run time | 29.359 | 31.785 | **+2.426** |
Daily averages run time | 1.187 | 0.979 | 0.208 |
Total execution time | 60.463 | 58.555 | 1.908 (3% faster) |

Again, differences **in bold** indicate that running with the minimised data is **slower**.

Now the overall runtime is very slightly faster than with the original data, but it's only a small change. Note that exposure estimation is still slower - but in testing across multiple runs of the pipeline, a number of the values here were quite variable, so sometimes exposure estimation was faster with the slimmed data, and sometimes the total execution time was slower with the slimmed data.

In summary then, slimming the data down has not made much difference to the runtime of the R API, but I think it's still a useful step in making the R code more efficient.

**Review by andrewgryan — CHANGES_REQUESTED** [review]
Good work! All those `*_label` columns always felt like duplicates of the integer equivalents that are easily recoverable at the end of the process.

To maintain consistency with the current behaviour, can we add a post-processing step that puts those columns back in at the end of the aggregation step, e.g. `daily_averages.R`?

**Review by andrewgryan — APPROVED** [review]
Actually, on second thoughts the label business logic is not a back-end concern at the moment

**Comment by DPeterK** [comment]
@andrewgryan fully agree on the labels front - but:
* they duplicate the content of the indices for each label unnecessarily, leading to carting around a lot more data per frame than we need, and
* they're for human consumption, not machine. As a human should never be viewing the DIMEX service, we shouldn't need them in the machine bit.

Then we use the frontend to translate between human-readable labels and machine indices, so that we never expose the indices to the human.
